mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -150,8 +166,26 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
             });
 
             if (apiResponse.type === 'success') {
+
                 const response = apiResponse.response;
                 const aiMessage = response.choices[0].message;
+
+                const aiGeneratedCode = getCodeBlockFromMessage(aiMessage);
+                const aiGeneratedCodeCleaned = removeMarkdownCodeFormatting(aiGeneratedCode || '');
+
+                const lineChanges = getCodeDiffLineRanges(activeCellCode, aiGeneratedCodeCleaned)
+
+                const unifiedDiffs = createUnifiedDiff(activeCellCode, aiGeneratedCodeCleaned, lineChanges)
+
+                const unifiedCodeString = addMarkdownCodeFormatting(unifiedDiffs.map(line => {
+                    return line.content !== undefined ? line.content : ''
+                }).join('\n'))
```

Comment:
I think it would be better to call a function like "getCodeDiffsAndUnifiedCode" that does all of this and returns both the unified code and unified diffs. I don't need to be doing all of this function calling here.